plugins/modules/win_dhcp_server.ps1

```diff
@@ -0,0 +1,259 @@
+$scope_id = $module.Params.scope
+$startrange = $module.Params.startrange
+$endrange = $module.Params.endrange
+$name = $module.Params.name
+$scopestate = $module.Params.scopestate # defines if the scope is activated or inactive
+#extraparam
+$state = $module.Params.state
+$computername = $module.Params.computername #optional to specifiy another computer
+$force = $module.Params.force #optional for forcing Parameter
+#Scope
+[System.TimeSpan]$validlifetime = $module.Params.validlifetime   #needed for IPv6 Scope
+[System.TimeSpan]$preferredlifetime = $module.Params.preferredlifetime #needed for IPv6 Scope
+$subnetmask = $module.Params.subnetmask #needed for IPv4 Scope
+$extra_param = @{}
+if($computername) { $extra_param.ComputerName = $computername }
+
+#ToDo: Need to validate each Param for types
+$varsnotcorrect = $true
+if($type -eq "scope" -and $version -eq "IPv6" -and $null -ne $scope_id -and $null -ne $preferredlifetime -and $null -ne $validlifetime) {
+    #type scope and IPV6
+    $varsnotcorrect = $false
+    if($preferredlifetime -gt $validlifetime) {
+        $varsnotcorrect = $true
+        $msg = "The Preferred LifeTime needs to be less or equal ValidLifetime"
+    }
+}
+if($type -eq "scope" -and $version -eq "IPv4" -and $null -ne $scope_id -and $null -ne $subnetmask -and $null -ne $startrange) {
+    #type scope and IPV6
+    #nothing to do here
+    $varsnotcorrect = $false
+}
+if($varsnotcorrect) {
+    #Exit Module when Vars not correctly specified
+    $module.FailJson("Please check your Variables. $msg")
+}
+
+Try {
+    # Import DHCP Server PS Module
+    Import-Module DhcpServer
+} Catch {
+    # Couldn't load the DhcpServer Module
+    $module.FailJson("The DhcpServer module failed to load properly: $($_.Exception.Message)", $_)
+}
+# State: Absent
+if ($state -eq "absent") {
+    #Remove scope
+    if ($type -eq "scope") {
+        $current_scope_exists = $false
+        if($version -eq "IPv4") {
+            $dhcpscopes = Get-DhcpServerv4Scope @extra_param
+            $current_scope = $dhcpscopes | Where-Object { $_.ScopeId -eq $scope_id } | Select-Object *
+            if ($current_scope) { $current_scope_exists = $true }
+        } else {
+            #IPv6
+            $current_scope = Get-DhcpServerv6Scope -Prefix $scope_id @extra_param
+            if ($current_scope.Prefix -contains $scope_id) { $current_scope_exists = $true }
+        }
+        if($check_mode) { $extra_param.WhatIf = $check_mode }
+        if($force) { $extra_param.Force = $force }
+        if ($current_scope_exists -eq $false) {
+            $module.Result.msg = "The scope doesn't exist."
+        } else {
+            # If the scope exists, we need to destroy it
+            if ($version -eq "IPv4") {
+                try {
+                    Remove-DhcpServerv4Scope -ScopeId $scope_id @extra_param
+                    $module.Result.changed = $true
+                } catch {
+                    $module.Result.changed = $false
```

Comment:
No need to set back to false and really no need for the catch block, just let the exceptions flow.